src/pages/myResult/[teamcode]/[userId].tsx

```diff
@@ -28,25 +30,40 @@ interface userIdType {
 function MyResult({ userId, teamId }: userIdType) {
   const [resultData, setResultData] = useState<UserData>();
   const [resultCharacter, setResultCharacter] = useState(0);
+  const [isVistor, setIsVistor] = useState(false);
   const { data } = useQuery('userData', () => getMyResult(userId));
-  const imgToDownload = 'downloadImg';
+  const [modalState, setModalState] = useState(false);
+  const { query, isReady } = useRouter();
   useEffect(() => {
     setResultData(data);
     const inputData = setConstantIndex(data?.result[4]?.questionType);
     setResultCharacter(inputData);
   }, [data]);
+  useEffect(() => {
+    if (!isReady) return;
+    if (query.teamcode === String(teamId)) {
+      setIsVistor(false);
+    } else {
+      setIsVistor(true);
+    }
+  }, [isReady]);
 
   return (
     <StmyResultPage>
       <SEO title="T.time | 팀과 내가 함께 성장하는 시간" description="결과를 확인해보세요!" />
       <LogoTop />
       {resultData ? (
         <StMyResult>
+          {modalState ? (
+            <MyResultModal userId={String(userId)} userName={resultData.nickname} setModalState={setModalState} />
+          ) : (
+            <></>
+          )}
```

Comment:
```suggestion
          {modalState && (
            <MyResultModal userId={String(userId)} userName={resultData.nickname} setModalState={setModalState} />
          )}
```
이렇게 && 연산자로 작성하면 코드를 줄일 수 있어효!